search-commons/src/main/java/no/unit/nva/search/common/ParameterValidator.java

```diff
@@ -177,15 +178,25 @@ public ParameterValidator<K, Q> fromRequestInfo(RequestInfo requestInfo) {
         searchQuery.setMediaType(requestInfo.getHeaders().get(ACCEPT));
         var uri = URI.create(HTTPS + requestInfo.getDomainName() + requestInfo.getPath());
         searchQuery.setNvaSearchApiUri(uri);
-        return fromQueryParameters(requestInfo.getQueryParameters());
+        return fromMultiValueParameters(requestInfo.getMultiValueQueryStringParameters());
+    }
+
+    /**
+     * Adds testParameters from query.
+     *
+     * @apiNote This is intended to be used in runtime
+     */
+    public ParameterValidator<K, Q> fromMultiValueParameters(Map<String, List<String>> parameters) {
+        parameters.forEach((k, v) -> v.forEach(value -> setValue(k, value)));
+        return this;
```

Comment:
I cant see a test for this. I suspect it might put/set the latest value, and not use anythign else. Would be great with a test that proves that this approach works

Reply:
Since fromMultiValueParameters is intended for runtime only, I updated a Handler test.